### ISSUE TITLE
Generalize REUniverse

### ISSUE THREAD

**Issue by theavey** [theavey/ParaTemp#13]
As implemented, this is really focused on parallel tempering-type replica exchange (where the temperatures differ between the replicas). 

It would be good to ask someone who does non-temperature replica exchange to figure out how it would be best to do the indexing.
There might be some other parameter that might be natural to index by.

At the very least, possibly `temps` could be renamed to `index` or similar, and made optional.
Then, there could be a new argument `index_type` possibly that if it is 'temp', the replica's temperatures can be set by index.

**Comment by theavey** [comment]
Hey @dstelter92, you have any insight into this?

I have an class this is a set of replica trajectories. I mostly index them by temperature, which is natural for me, but do you have any thoughts on a more general way to index?

I guess the most general would just be an index, but that could take away by fun of being able to access one with `REUniverse[temperature]`.

**Comment by theavey** [comment]
So just essentially some arbitrary index that you keep track of using `temp2world`?

That seems very general: just use some index, but then define helper functions to go from some useful metric to the general index.

**Comment by dstelter92** [comment]
Yep. In the case of running the simulation, the index corresponds to the MPI world in the MPI universe. In your case for post-processing, it would be arbitrary.
